### PR TITLE
[feat/aut819] Implement version bumper

### PR DESCRIPTION
**Pull request by ChristopherAbram**
+ Provide script for reading and updating smartpeak version from CMake configuration (smartpeak tool python package)
+ Provide scripts for bumping/setting smartpeak version from commit message
+ Update smartpeak version to the latest released

Note: I placed the package in the `tools` subdirectory.

### CODE REVIEW

File: tools/smartpeak/smartpeak_git_prepare_release [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+        # Read current version:
+        config_file = os.path.join(utils.get_project_path(), 'CMakeLists.txt')
+        major, minor, patch = utils.get_version(config_file)
+
+        # Retrieve new version:
+        success, (n_major, n_minor, n_patch) = utils.get_new_version(
+            set_version, bump, (major, minor, patch))
+        if not success:
+            return 1
```
Comment (bertrandboudaud) [inline tools/smartpeak/smartpeak_git_prepare_release:21]: We don't need any error message?

File: tools/smartpeak/smartpeak_git_prepare_release [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      
+        # Update config:
+        success = utils.update_version(config_file, version=(n_major, n_minor, n_patch))
+        if not success:
+            return 1
```
Comment (bertrandboudaud) [inline tools/smartpeak/smartpeak_git_prepare_release:26]: We don't need any error message?

**Review by bertrandboudaud — APPROVED** [review]
looks great. ok for me

File: tools/smartpeak/smartpeak_git_prepare_release (hunk 1 above)
Comment (ChristopherAbram) [reply tools/smartpeak/smartpeak_git_prepare_release:21]: There are error messages in inside methods, just return code has to go be according in main script

File: tools/smartpeak/smartpeak_prepare_release [hunk 3]
```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/python
+from __future__ import print_function
+import sys, os, re, argparse
+
+import smartpeak.utils as utils
+
+
```
Comment (ghost) [inline tools/smartpeak/smartpeak_prepare_release:7]: It'd be nice to have some documentation for each function implemented, overall very nice.

Comment (ChristopherAbram) [reply tools/smartpeak/smartpeak_prepare_release:7]: Good point, I will add this while adding developer documentation

**Review by dmccloskey — CHANGES_REQUESTED** [review]
Looks very nice.

If you can 1) add in docstrings to the methods, 2) provide unit tests for the new methods, and 3) take care of the DeepCode issues, that would be great.

**Review by dmccloskey — CHANGES_REQUESTED** [review]
Docstrings and unit tests look great.  Thanks for adding those in.  The only thing I see left is adding the running of the unit tests into the CircleCI pipelines.  Ping me when it is ready and we can merge.

File: tools/smartpeak/smartpeak_send_emails [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+</div>
+<br><br>
+Best wishes,<br>
+SmartPeak Team
+<br><br>
+<span style="color: #999">This is an automatic notification. Please do not replay to this message.</span>'''.format(version=version, changes=changes)
+
+      msgPlain = '''\
+SmartPeak v{version} is released!\n
+We are constantly working to improve your SmartPeak experience, below a summary of what has changed in the new release.\n
```
Comment (dmccloskey) [inline tools/smartpeak/smartpeak_send_emails:142]: Very nice.

File: tools/smartpeak/smartpeak_send_emails [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+from email.mime.text import MIMEText
+import markdown
+
+import smartpeak.utils as utils
+
+# If modifying these scopes, delete the file token.pickle.
+SCOPES = ['https://www.googleapis.com/auth/gmail.send']
+
+
+def create_message(sender, to, subject, msgHtml, msgPlain):
```
Comment (dmccloskey) [inline tools/smartpeak/smartpeak_send_emails:19]: Did you add in unit tests for these new methods?

**Review by dmccloskey — CHANGES_REQUESTED** [review]
Additions look very nice.  My only minor comment the missing unit tests for the new python methods for the email functionality.

File: tools/smartpeak/smartpeak_send_emails (hunk 5 above)
Comment (ChristopherAbram) [reply tools/smartpeak/smartpeak_send_emails:19]: There is not really anything to test here, there two helper functions which only wraps message in MIME object and then content is encoded with base64. I would leave it as it is for now.

**Review by dmccloskey — APPROVED** [review]

**Pull request merged**